packages/core-plugin-api/src/analytics/Tracker.ts

```diff
@@ -43,13 +61,69 @@ export class Tracker implements AnalyticsTracker {
       attributes,
     }: { value?: number; attributes?: AnalyticsEventAttributes } = {},
   ) {
+    // Never pass internal "_element" context value.
+    const { _element, ...context } = this.context;
+
+    // Never fire the special "_routable-extension-rendered" internal event.
+    if (action === '_routable-extension-rendered') {
+      // Instead, push it onto the global store.
+      routableExtensionRenders.push({
+        context: {
+          ...context,
+          extension: 'App',
+        },
+      });
+      return;
+    }
+
+    // If we are about to fire a real event, and we have an un-fired gathered
+    // mountpoint navigation on the global store, we need to fire the navigate
+    // event first, so this real event happens accurately after the navigation.
+    if (gatheredNavigations.length) {
+      // Combine the most recent info from each.
+      const lastGatheredNavigation = gatheredNavigations.pop()!;
+      const lastRoutableRender = routableExtensionRenders.pop();
+
+      try {
+        this.analyticsApi.captureEvent({
+          ...lastGatheredNavigation,
+          ...lastRoutableRender,
+        });
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn('Error during analytics event capture. %o', e);
+      }
+
+      // Clear the global stores.
+      gatheredNavigations.length = 0;
+      routableExtensionRenders.length = 0;
+    }
+
+    // Never directly fire a navigation event on a gathered route with default
+    // contextual details.
+    if (
+      action === 'navigate' &&
+      _element === 'gathered' &&
+      context.pluginId === 'root'
+    ) {
+      // Instead, push it onto the global store.
+      gatheredNavigations.push({
+        action,
+        subject,
+        value,
+        attributes,
+        context,
+      });
```

Comment:
Ah, so as long as the user doesn't close the page at this point, we'll generally always emit a more accurate event later? Regardless of whether the user interacts on the page or navigates away, we'll end up with the gathered navigation being emitted later.

So the tradeoff with this approach is that we might drop some events if the page is closed, and we might have some situations where the next even happens before we're able to capture the rendered extension, so there might be a bit of flakiness in some analytics events wrt their reported plugin ID?

Reply:
> _Ah, so as long as the user doesn't close the page at this point, we'll generally always emit a more accurate event later? Regardless of whether the user interacts on the page or navigates away, we'll end up with the gathered navigation being emitted later._

Yep.  We could even mitigate for the "close the page" scenario by adding a `beforeunload` event handler?